### ISSUE TITLE
[ENH] Increase the Backtest Performance of models that are not updated

### ISSUE THREAD

**Issue by benHeid** [sktime/sktime#4650]
**Is your feature request related to a problem? Please describe.**
If performing deep learning-based time series forecasting (and with some other ml models), often the update step is omitted in production due to
* often costly training/retraining, and
* The catastrophic interference in neural networks.
Based on these considerations, we could implement faster and more efficient `update_predict` methods if we use this method for backtesting. 

**Describe the solution you'd like**
I would like a solution that creates samples for neural networks in the `update_predict` method and then pass all samples at once to the underlying model to exploit the often implemented parallelization in these models.

**Describe alternatives you've considered**
Leave it as it is and require a lot of time for backtesting non-updated models.

** Open Questions**
* Are there existing models that could benefit from such backtesting strategies? E.g., sklearn-based regressions?

**Comment by benHeid** [comment]
@fkiraly fyi

**Comment by fkiraly** [comment]
hm, my solution would be:

* in the respective model, ensure that `update_predict` carries out the tabulation step
* in `evaluate`, use the option that internally calls `update_predict`, i.e., `strategy="update"`

Now, currently, it is not clear in the extender contract how to implement efficient `update_predict` internally. This is a bit ot duct tape code at the moment, it should call an internal `_update_predict` where the fast `update_predict` logic would live.

Related issues:

https://github.com/sktime/sktime/issues/982 - general design question
https://github.com/sktime/sktime/issues/2118 - current extender contract has odd, undocumented corner case properties
https://github.com/sktime/sktime/issues/2669 - another instance where we could do `update_predict` much more efficiently
https://github.com/sktime/sktime/issues/4217 - also hinges on a clean `update_predict`
https://github.com/sktime/sktime/pull/3447 - draft PR I got distracted from, was planning to get back when there's less admin

**Comment by fkiraly** [comment]
(added the API design label due to the design question in #982 related)
